boards/arm/nrf9160dk_nrf52840/board.c

```diff
@@ -1,203 +1,85 @@
 /*
- * Copyright (c) 2018 Nordic Semiconductor ASA.
+ * Copyright (c) 2018-2021 Nordic Semiconductor ASA.
  *
  * SPDX-License-Identifier: Apache-2.0
  */
 
 #include <zephyr.h>
 #include <init.h>
 #include <drivers/gpio.h>
+#include <devicetree.h>
 #include <logging/log.h>
 
 LOG_MODULE_REGISTER(board_control, CONFIG_BOARD_NRF9160DK_LOG_LEVEL);
 
-/* The following pins on the nRF52840 control the routing of certain
- * components/lines on the nRF9160 DK. They are specified as follows:
- *
- * COMPONENT_SWITCH : ROUTING PIN
- *
- * NOTE: UART1_VCOM_U7 is on pin 12 of both P0 and P1.
- * Both P0.12 -and- P1.12 need to be toggled to route UART1 to VCOM2.
- */
-
-/* GPIO pins on Port 0 */
-
-#define INTERFACE0_U5 13 /* MCU interface pins 0 - 2 */
-#define INTERFACE1_U6 24 /* MCU interface pins 3 - 5 */
-#define UART1_VCOM_U7 12 /* Route nRF9160 UART1 to VCOM2 */
-#define BUTTON1_U12 6
-#define BUTTON2_U12 26
-#define SWITCH2_U9 8
-
-/* GPIO pins on Port 1 */
-
-#define INTERFACE2_U21 10 /* COEX interface pins 6 - 8 */
-#define UART0_VCOM_U14 14 /* Route nRF9160 UART0 to VCOM0 */
-#define UART1_VCOM_U7 12  /* Route nRF9160 UART1 to VCOM2 */
-#define LED1_U8 5
-#define LED2_U8 7
-#define LED3_U11 1
-#define LED4_U11 3
-#define SWITCH1_U9 9
-
-/* MCU interface pins
- * These pins can be used for inter-SoC communication.
- *
- * | nRF9160 |				 | nRF52840 | nRF9160 DK |
- * | P0.17   | -- MCU Interface Pin 0 -- | P0.17    | Arduino 4  |
- * | P0.18   | -- MCU Interface Pin 1 -- | P0.20    | Arduino 5  |
- * | P0.19   | -- MCU Interface Pin 2 -- | P0.15    | Arduino 6  |
- * | P0.21   | -- MCU Interface Pin 3 -- | P0.22    | TRACECLK   |
- * | P0.22   | -- MCU Interface Pin 4 -- | P1.04    | TRACEDATA0 |
- * | P0.23   | -- MCU Interface Pin 5 -- | P1.02    | TRACEDATA1 |
- * | COEX0   | -- MCU Interface Pin 6 -- | P1.13    | COEX0_PH   |
- * | COEX1   | -- MCU Interface Pin 7 -- | P1.11    | COEX1_PH   |
- * | COEX2   | -- MCU Interface Pin 8 -- | P1.15    | COEX2_PH   |
- */
-
-__packed struct pin_config {
-	uint8_t pin;
-	uint8_t val;
-};
-
-/* The following tables specify the configuration of each pin based on the
- * Kconfig options that drive it.
- * The switches have active-low logic, so when writing to the port we will
- * need to invert the value to match the IS_ENABLED() logic.
- */
-
-static const struct pin_config pins_on_p0[] = {
-	{ INTERFACE0_U5, IS_ENABLED(CONFIG_BOARD_NRF9160DK_INTERFACE0_ARDUINO) },
-	{ INTERFACE1_U6, IS_ENABLED(CONFIG_BOARD_NRF9160DK_INTERFACE1_TRACE) },
-	{ UART1_VCOM_U7, IS_ENABLED(CONFIG_BOARD_NRF9160DK_UART1_ARDUINO) },
-	{ BUTTON1_U12,   IS_ENABLED(CONFIG_BOARD_NRF9160DK_BUTTON0_PHY) },
-	{ BUTTON2_U12,   IS_ENABLED(CONFIG_BOARD_NRF9160DK_BUTTON1_PHY) },
-	{ SWITCH2_U9,    IS_ENABLED(CONFIG_BOARD_NRF9160DK_SWITCH1_PHY) },
+#define GET_CTRL(name, prop, idx) \
```

Comment:
Might want to call this one GET_CTLR instead, for consistency.

Reply:
Of course, that was my intention. I just made a typo here.